### ISSUE TITLE
Why doesn't the AST define an instance for `Eq`?

### ISSUE THREAD

**Issue by ivanperez-keera** [fdedden/language-c99-simple#16]
I just noticed that the AST does not define an instance for `Eq`.

It would be useful to be able to compare AST branches for equality (for example, to ensure you are not generating the same expression twice, or to be able to put declarations in a set). Is there a fundamental reason why it cannot or should not be done
?

**Comment by fdedden** [comment]
No, there should not be a specific reason for omitting the `Eq` instance. So far there never was a need for it.

We can add the instance, and probably add it for `language-c99` as well.
